pymeasure/instruments/oxfordinstruments/ips120_10.py

```diff
@@ -0,0 +1,470 @@
+                                    # current is ramped back to 0 (i.e. persistent mode).
+
+        print(self.field)           # Print the current field (whether in persistent or
+                                    # non-persistent mode)
+
+        ips.set_field(0)            # Bring the magnet to 0 T. The persistent mode will be
+                                    # turned off first (i.e. current back to set-point and
+                                    # switch-heater on); afterwards the switch-heater will
+                                    # again be turned off.
+
+        ips.disable_control()       # Disables the control of the supply, turns off the
+                                    # switch-heater and clamps the output.
+
+    :param clear_buffer: A boolean property that controls whether the instrument
+        buffer is clear upon initialisation.
+    :param switch_heater_heating_delay: The time in seconds (default is 20s) to wait after
+        the switch-heater is turned on before the heater is expected to be heated.
+    :param switch_heater_cooling_delay: The time in seconds (default is 20s) to wait after
+        the switch-heater is turned off before the heater is expected to be cooled down.
+    :param field_range: A numeric value or a tuple of two values to indicate the
+        lowest and highest allowed magnetic fields. If a numeric value is provided
+        the range is expected to be from :code:`-field_range` to :code:`+field_range`.
+
+    """
+
+    _MAX_CURRENT = 120  # Ampere
+    _MAX_VOLTAGE = 10  # Volts
+    _SWITCH_HEATER_HEATING_DELAY = 20  # Seconds
+    _SWITCH_HEATER_COOLING_DELAY = 20  # Seconds
+    _FIELD_RANGE = [-7, 7]  # Tesla
+
+    _SWITCH_HEATER_SET_VALUES = {
+        False: 0,  # Heater off
+        True: 1,  # Heater on, with safety checks
+        "Force": 2,  # Heater on, without safety checks
+    }
+    _SWITCH_HEATER_GET_VALUES = {
+        0: False,  # Heater off, Switch closed, Magnet at zero
+        1: True,  # Heater on, Switch open
+        2: False,  # Heater off, Switch closed, Magnet at field
+        5: "Heater fault, low heater current",  # Heater on but current is low
+        8: "No switch fitted",  # No switch fitted
+    }
+
+    def __init__(self, resourceName,
+                 clear_buffer=True,
+                 switch_heater_heating_delay=None,
+                 switch_heater_cooling_delay=None,
+                 field_range=None,
+                 **kwargs):
+
+        super().__init__(
+            resourceName,
+            "Oxford IPS",
+            includeSCPI=False,
+            send_end=True,
+            read_termination="\r",
+            **kwargs
+        )
+
+        if isinstance(self.adapter, VISAAdapter):
+            self.adapter.connection.baud_rate = 9600
+            self.adapter.connection.data_bits = 8
+            self.adapter.connection.parity = 0
+            self.adapter.connection.stop_bits = 20
+
```

Comment:
ah, this should be updated accounting for the merged #539:
```suggestion
        kwargs.setdefault('read_termination', '\r')
        super().__init__(
            resourceName,
            "Oxford IPS",
            includeSCPI=False,
            gpib={'send_end': True},
            asrl={'baud_rate': 9600,
                  'data_bits': 8,
                  'parity': 0,
                  'stop_bits': 20,
                  },
            **kwargs
        )

```